packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1076,8 +1185,60 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
     }
   }
 
-  @override
-  Widget build(BuildContext context) {
+  Widget buildCupertinoTextField(BuildContext context) {
+    return CupertinoTextField(
+      key: widget.key,
```

Comment:
I didn't check all the fields. Take a look to make sure for any fields that is used in one implementation and not the other, mention so in that field's documentation.